libraries/botbuilder-ai/botbuilder/ai/luis/luis_recognizer_v3.py

```diff
@@ -0,0 +1,176 @@
+import aiohttp
+import asyncio
+import json
+
+from typing import Dict
+from .luis_recognizer_internal import LuisRecognizerInternal
+from .luis_recognizer_options_v3 import LuisRecognizerOptionsV3
+from .luis_application import LuisApplication
+
+from botbuilder.core import (
+    RecognizerResult,
+    TurnContext,
+)
+from .activity_util import ActivityUtil
+
+
+class LuisRecognizerV3(LuisRecognizerInternal):
+    _dateSubtypes = ["date", "daterange", "datetime", "datetimerange", "duration", "set", "time", "timerange"]
+    _geographySubtypes = ["poi", "city", "countryRegion", "continent", "state"]
+    _metadata_key = "$instance"
+    # The value type for a LUIS trace activity.
+    luis_trace_type: str = "https://www.luis.ai/schemas/trace"
+
+    # The context label for a LUIS trace activity.
+    luis_trace_label: str = "Luis Trace"
+
+    def __init__(self, luis_application: LuisApplication, luis_recognizer_options_v3: LuisRecognizerOptionsV3 = None):
+        super().__init__(luis_application)
+
+        self.luis_recognizer_options_v3 = luis_recognizer_options_v3 or LuisRecognizerOptionsV3()
+        self._application = luis_application
+
+    async def recognizer_internal(
+            self,
+            turn_context: TurnContext):
+        recognizer_result: RecognizerResult = None
+
+        utterance: str = turn_context.activity.text if turn_context.activity is not None else None
+
+        url = self._build_url()
+        body = self._build_request(utterance)
```

Comment:
This probably doesn't need to be changed, but I would have had _build_request return a tuple of body, headers.  This consolidates that logic into a single method.

Used like:
body, headers = self._build_request(utterance)

Reply:
I will keep this for now until I refactor the other sdks to keep parity